ca/django_ca/utils.py

```diff
@@ -209,8 +214,11 @@ def getter(self, method):  # type: ignore
 
 
 def sort_name(subject: List[Tuple[str, str]]) -> List[Tuple[str, str]]:
-    """Returns the subject in the correct order for a x509 subject."""
-    return sorted(subject, key=lambda e: SUBJECT_FIELDS.index(e[0]))
+    """Returns the subject in the correct order for a x509 subject, while respecting
+    the original list order for possible subject fields allowing for MULTIPLE_OIDS."""
+    if SUBJECT_FIELDS.index(subject[0][0]) > SUBJECT_FIELDS.index(subject[-1][0]):
+        return sorted(subject[::-1], key=lambda k: SUBJECT_FIELDS.index(k[0]))
+    return sorted(subject, key=lambda k: SUBJECT_FIELDS.index(k[0]))
```

Comment:
This function here: I _think_ I understand what your updated comment means to express, but the code is far from obvious. And I'm frankly surprised that the old code does not respect sort stability.

1. I think `subject[::-1]` is just reversing the list. At least a simple test did that. If yes, then use the more obvious `reversed()` instead.
2. What does that `if` check do? 
3. Definitely add tests for an updated version of this function.

Reply:
See comment below. I'm not sure how it dies it anymore, I just know it yields the desired results.

Exemplary test cases as mentioned below:

```python
>>> subject = [('CN', 'John Doe'), ('OU', 'Users'), ('DC', 'contoso'), ('DC', 'loc')] 
>>> sort_name(subject)
[('DC', 'loc'), ('DC', 'contoso'), ('OU', 'Users'), ('CN', 'John Doe')]

>>> subject = [('CN', 'John Doe'), ('OU', 'Marketing'), ('OU', 'Users'), ('OU', 'OtherUnit'), ('DC', 'contoso'), ('DC', 'loc')]
>>> sort_name(subject)
[('DC', 'loc'), ('DC', 'contoso'), ('OU', 'OtherUnit'), ('OU', 'Users'), ('OU', 'Marketing'), ('CN', 'John Doe')]

>>> subject = [('DC', 'loc'), ('DC', 'contoso'), ('DC', 'ad'), ('OU', 'Users'), ('CN', 'John Doe')]
>>> sort_name(subject)
[('DC', 'loc'), ('DC', 'contoso'), ('DC', 'ad'), ('OU', 'Users'), ('CN', 'John Doe')]
```

Let me know if you feel comfortable with the sort function so we can move forward. Halting any further effort until then.

Reply:
Now that's behavior that I don't understand: In testcase two, `loc` is the last DC in the input and ends up as first element.

In the third test case, `loc` is the _first_ DC, and still ends up as last.

Reply:
Yes. In the first and second test case, the user input the subject with `CN` being the first element – this is a classic case of assuming the hierarchy goes from left-to-right with `CN` being the lowest weighted element.

In test case three, we're looking at the opposite. The user's hierarchical input is correct as per `X509` with `CN` being the last element.

In all three test cases the user respect the general hierarchical-structure, albeit in a reversed manner.